apps/studio/src/components/ResourceSelector/useResourceStack.tsx

```diff
@@ -38,9 +40,17 @@ export const useResourceStack = ({
     [resourceStack],
   )
 
-  const removeFromStack = useCallback((numberOfResources: number): void => {
-    setResourceStack((prev) => prev.slice(0, -numberOfResources))
-  }, [])
+  const removeFromStack = useCallback(
+    (numberOfResources: number): ResourceItemContent[] => {
+      let updatedStack: ResourceItemContent[] = []
+      setResourceStack((prev) => {
+        updatedStack = prev.slice(0, -numberOfResources)
+        return updatedStack
+      })
+      return updatedStack
+    },
+    [],
+  )
```

Comment:
[question] is this just to return the new state of the stack? curious why we don't just return `prev` directly so it's easier to understand (but not a huge issue since this part of the codebase is unlikely to change)